components/script/webdriver_handlers.rs

```
@@ -245,7 +246,22 @@ pub unsafe fn jsval_to_webdriver(
                 Err(WebDriverJSError::UnknownType)
             }
         } else {
-            Err(WebDriverJSError::UnknownType)
+            let mut result = HashMap::new();
+
+            let common_properties = vec!["x", "y", "width", "height", "key"];
+            for property in common_properties.iter() {
+                rooted!(in(cx) let mut item = UndefinedValue());
+                if let Ok(_) = get_property_jsval(cx, object.handle(), property, item.handle_mut())
```

We should throw a DOM exception if this fails.